elasticsearch/examples/upgrade/scripts/upgrade.sh

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env bash
+
+set -eo pipefail
+
+usage() {
+  cat <<-EOF
+	USAGE:
+	  $0 [--release <release-name>] [--from <elasticsearch-version>] --to <elasticsearch-version>
+	  $0 --help
+
+	OPTIONS:
+    --release <release-name>
+	    Name of the Helm release to install
+	  --from <elasticsearch-version>
+	    Elasticsearch version to use for first install
+	EOF
+  exit 1
+}
+
+RELEASE="helm-es-upgrade"
+FROM=""
+
+while [[ $# -gt 0 ]]
+do
+  key="$1"
+
+  case $key in
+    --help)
+      usage
+    ;;
+    --release)
+      RELEASE="$2"
+      shift 2
+    ;;
+    --from)
+      FROM="$2"
+      shift 2
+    ;;
+    *)
+      log "Unrecognized argument: '$key'"
+      usage
+    ;;
+  esac
+done
+
+# Elasticsearch chart < 7.4.0 are not compatible with K8S >= 1.16)
+if [[ -z $FROM ]]
+then
+  KUBE_MINOR_VERSION=$(kubectl version --client -o yaml | grep minor | sed 's/[^0-9]*//g')
```

Comment:
I think we don't want `--client` since that will show the version of `kubectl` that is being used, which isn't necessarily the server version (especially if someone is trying this out locally), getting the server version would be more accurate.

Re-examining how these jobs and makefiles fit together, I wonder if instead we should pass along the `KUBERNETES_VERSION` that is being passed in by the entrypoint from the jenkins job
https://github.com/elastic/helm-charts/blob/4ece77dd1685ad5a787da9070dc6099417810e53/.ci/jobs/elastic%2Bhelm-charts%2Bstaging%2Bintegration-elasticsearch.yml#L54

Where we could pass it along by updating this to `make KUBERNETES_VERSION=$(KUBERNETES_VERSION)`
https://github.com/elastic/helm-charts/blob/4ece77dd1685ad5a787da9070dc6099417810e53/helpers/terraform/Makefile#L91

Then we could pass this script the KUBERNETES_VERSION with an option like `--kubernetes-version` and determine the elasticsearch version based on that, WDYT?

Reply:
After re-thinking about this, I don't like it, I think it is better to figure out the real server version using `kubectl` somehow.

Reply:
And I think `kubectl ... -o json | jq` is the best way to accomplish it in a script despite it becoming a dependency.